code/game/objects/items/storage/firstaid.dm

```diff
@@ -501,3 +504,31 @@
 	req_access = list(ACCESS_NT_CORPORATE)
 	req_id_role = CORPORATE_LIAISON
 	scan_name = TRUE
+
+/obj/item/storage/pill_bottle/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(.)
+		return
+	if(!istype(I, /obj/item/facepaint) || istype(src, /obj/item/storage/pill_bottle/packet))
+		return
+
+	var/obj/item/facepaint/paint = I
+	if(paint.uses < 1)
+		to_chat(user, span_warning("\the [paint] is out of color!"))
+		return
+	var/new_color
+	var/new_color2
+	if(colorable_colors)
+		new_color = colorable_colors[tgui_input_list(user, "Pick a color", "Pick color", colorable_colors)]
+		new_color2 = colorable_colors[tgui_input_list(user, "Pick a color", "Pick color", colorable_colors)]
+	else
+		new_color = input(user, "Pick a color", "Pick color") as null|color
+		new_color2 = input(user, "Pick a color", "Pick color") as null|color
+
```

Comment:
I wnt make you but if you see helmet visor coloring code it uses attackby and attackby alternate to make it so you color the helmet with left click and the visor with right, admittedly its bugged but it would be a nice qol if you want to do it

Reply:
which line is attack by alternate on of which file